src/frontend/src/pages/ProjectDetailPage/ProjectViewContainer/TaskListTabPanel.tsx

```diff
@@ -3,28 +3,197 @@
  * See the LICENSE file in the repository root folder for details.
  */
 
-import { Box, Link, Typography } from '@mui/material';
+import { Box, IconButton, Link, ListItemIcon, Menu, MenuItem, Typography } from '@mui/material';
+import { DataGrid, GridColDef, GridColumnHeaderParams } from '@mui/x-data-grid';
+import { RoleEnum, Task, TaskStatus, UserPreview } from 'shared';
+import { datePipe, userPreviewFullNamePipe } from '../../../utils/pipes';
+import { GridColDefStyle } from '../../../utils/tables';
+import MenuIcon from '@mui/icons-material/Menu';
+import { useState } from 'react';
+import PauseIcon from '@mui/icons-material/Pause';
+import DeleteIcon from '@mui/icons-material/Delete';
+import PlayArrowIcon from '@mui/icons-material/PlayArrow';
+import CheckIcon from '@mui/icons-material/Check';
+import { useAuth } from '../../../hooks/auth.hooks';
+import LoadingIndicator from '../../../components/LoadingIndicator';
 
 interface TaskListTabPanelProps {
   index: number;
   value: number;
+  tasks: Task[];
+  status: TaskStatus;
 }
 
 const TaskListTabPanel = (props: TaskListTabPanelProps) => {
-  const { value, index } = props;
+  const { value, index, tasks, status } = props;
+  const [anchorEl, setAnchorEl] = useState<null | HTMLElement>(null);
+
+  const auth = useAuth();
+
+  if (!auth.user) return <LoadingIndicator />;
+
+  const disabled = auth.user.role === RoleEnum.GUEST;
+
+  const handleMenu = (event: React.MouseEvent<HTMLElement>) => setAnchorEl(event.currentTarget);
+  const handleClose = () => setAnchorEl(null);
 
   // Skeleton copied from https://mui.com/material-ui/react-tabs/.
   // If they release the TabPanel component from @mui/lab to @mui/material then change the div to TabPanel.
 
+  const renderActionButtons = () => (
```

Comment:
I see no reason why we should be doing all this custom instead of using default data grid actions column stuff